__tests__/storage/storage.test.js

```diff
@@ -708,3 +709,19 @@ describe('checkPin and checkPassword', () => {
     await expect(storage.checkPassword('0000')).rejects.toThrow();
   }
 });
+
+test('isHardware', async () => {
+  const store = new MemoryStore();
+  const storage = new Storage(store);
+  const accessDataSpy = jest.spyOn(storage, '_getValidAccessData');
+
+  accessDataSpy.mockReturnValue(Promise.resolve({
+    walletFlags: WALLET_FLAGS.HARDWARE,
+  }));
+  await expect(storage.isHardwareWallet()).resolves.toBe(true);
+
+  accessDataSpy.mockReturnValue(Promise.resolve({
+    walletFlags: 0,
```

Comment:
Maybe add a test with READONLY flag to validate it will return false as well.

Reply:
Done